app/src/adapters/db/client.py

```diff
@@ -37,16 +32,17 @@ class DBClient:
     It has methods for getting a new connection or session object.
     """
 
+    _db_engine: DbEngine
     _engine: sqlalchemy.engine.Engine
 
-    def __init__(self) -> None:
-        self.config = get_db_config()
-        self._engine = _create_db_engine(self.config)
+    def __init__(self, db_engine: DbEngine) -> None:
```

Comment:
Let's assume this works relatively easily, what is more intuitive to understand as a developer?

I'd argue having explicitly defined DB engines makes more sense to me. "I have session A that works with DB A, and session B that works with DB B" rather than a single session that works under the hood to do it.

I think anything to try and make this work requires it's own dedicated investigation, and we'd want multiple DBs setup to be able to test and validate the behavior, which we don't currently have (in NJ, this is prep work for eventually having that). I'm just trying to get the configuration setup to make multiple DBClients possible right now, I don't have the bandwidth to go beyond that right now.

Reply:
yeah, i think our initial approach was easier to understand how it works, and until we see a more complex use case it's not worth overengineering it